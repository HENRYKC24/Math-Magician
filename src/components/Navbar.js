import React from 'react';
import Li from './Li';

const Navbar = () => {
  const routes = [
    { route: '/', name: 'Home' },
    { route: '/calculator', name: 'Calculator' },
    { route: '/quotes', name: 'Quotes' },
  ];

  return (
    <nav className="navbar">
      <h1 className="app-name">Math Magicians</h1>
      <ul className="nav-items">
        {routes.map((route, index) => (
          <Li
            key={Math.random()}
            index={index}
            name={route.name}
            route={route.route}
          />
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
