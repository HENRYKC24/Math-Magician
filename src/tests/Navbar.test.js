import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

it('Navbar renders the UI as expected', () => {
  const tree = renderer.create(<Router>
    <Navbar />
  </Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
