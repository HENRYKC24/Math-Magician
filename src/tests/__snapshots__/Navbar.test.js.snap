// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the UI as expected 1`] = `
<nav
  className="navbar"
>
  <h1
    className="app-name"
  >
    Math Magicians
  </h1>
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current="page"
        className="item active-link"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    |
    <li>
      <a
        aria-current={null}
        className="item"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    |
    <li>
      <a
        aria-current={null}
        className="item"
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
